src/data/quot.lean

```diff
@@ -496,4 +496,14 @@ noncomputable def out' (a : quotient s₁) : α := quotient.out a
 theorem mk_out' (a : α) : @setoid.r α s₁ (quotient.mk' a : quotient s₁).out' a :=
 quotient.exact (quotient.out_eq _)
 
+variables {α' : Type*} (r : α' → α' → Prop )
+
+lemma subsingleton_quot_iff_forall : subsingleton (quot r) ↔ ∀ a b, eqv_gen r a b :=
+begin
+  rw subsingleton_iff,
```

Comment:
I guess this isn't defined yet, it's from

https://github.com/leanprover-community/mathlib/blob/62de5912a37603789cdcdcbef4488826ca416627/src/logic/nontrivial.lean#L89

Perhaps we should move that to `logic.basic`? Or maybe this lemma belongs in `logic.nontrivial`?

Thoughts @digama0, @jcommelin?

Reply:
So I just put it onto logic.basic for the moment.  The proof of  `subsingleton_quot_iff_forall` is now complaining about the `forall` saying ``invalid field notation, 'forall' is not a valid "field" because environment does not contain 'function.surjective.forall'
  surjective_quot_mk ?m_2
which has type
  function.surjective (quot.mk ?m_2)``
 I'm not quite sure how to fix this (other than rewriting the proof).

Reply:
This is because `function.surjective.forall` probably comes from another file.

Reply:
Maybe best to ask Zulip for help on where to put this lemma.

Reply:
Sure I'll do that :)

Reply:
After asking I put it in `data/setoid`.